ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/ContractsReader.scala

```diff
@@ -55,6 +63,45 @@ private[dao] sealed abstract class ContractsReader(
           )
       })
 
+  /** Lookup of a contract in the case the contract value is already known (loaded from a cache) */
+  private[this] def lookupActiveContractWithoutArgument(
+      submitter: Party,
+      contractId: ContractId,
+      createArgument: Value,
+  ): Future[Option[Contract]] =
+    dispatcher
+      .executeSql(metrics.daml.index.db.lookupActiveContractWithoutArgumentDbMetrics) {
+        implicit connection =>
+          SQL"select participant_contracts.contract_id, template_id from #$contractsTable where contract_witness = $submitter and participant_contracts.contract_id = $contractId"
+            .as(contractWithoutValueRowParser.singleOpt)
+      }
+      .map(
+        _.map(
+          templateId =>
+            toContract(
+              contractId = contractId,
+              templateId = templateId,
+              createArgument = createArgument,
+          )))
+
+  override def lookupActiveContract(
+      submitter: Party,
+      contractId: ContractId,
+  ): Future[Option[Contract]] =
+    // Depending on whether the contract argument is cached or not, submit a different query to the database
+    translation.cache.getIfPresent(LfValueTranslation.Cache.ContractKey(contractId.coid)) match {
+      case Some(createArgument) =>
+        metrics.daml.index.db.lookupActiveContractArgumentCache.hitCount.inc()
```

Comment:
The cache is already instrumented. Do we have some other way of recording hits and misses for contracts?

Reply:
The cache is instrumented, but it only measures an overall hit/miss rate. I wanted to specifically measure the hit rate for contract arguments during active contract lookups.

Since this is not a very important metric, we could also remove it.

Reply:
I don't know, I recognize the importance of having this in to measure the impact. If I'm not mistaken we are actually using the same cache to store both creates for events and contracts, so they are effectively stored twice. If that's true, would it not make more sense to simply store them separately?